metamist/parser/generic_metadata_parser.py

```diff
@@ -474,27 +487,26 @@ def prepare_dict_from_keys(key_parts: list[str], val):
         dicts = []
         for row_key, dict_key in key_map.items():
             if isinstance(row, list):
-                inner_values = [r[row_key] for r in row if r.get(row_key) is not None]
+                inner_values = [string_to_bool(r[row_key]) for r in row if r.get(row_key) is not None]
                 if any(isinstance(inner, list) for inner in inner_values):
                     # lists are unhashable
                     value = inner_values
                 else:
-                    value = list(set(inner_values))
+                    value = sorted(list(set(inner_values)), key=str)
```

Comment:
You can sorted the set directly without the list. I don't see a clear reason why you'd sort, but I don't think it's bad to either. Can you add a comment why you'd want it sorted?

```suggestion
                    value = sorted(set(inner_values), key=str)
```

Reply:
Cheers! Yeah this was causing issues when I was running the tests. Because the collapsed meta becomes a list, the ordering was random making the test result 50/50 if it would pass or not. Decided the sorting was necessary to get consistent test results. I'll add a comment :)